From aa4166d0349224aa38f01897054429da8a3bdb32 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 16 Sep 2020 14:36:44 +0000
Subject: GCC: do not make pdf::Thumbnail move constructor/assignment operator
 noexcept
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GCC fails to build with this:
../../pdf/thumbnail.cc:116:1: error: function ‘chrome_pdf::Thumbnail::Thumbnail(chrome_pdf::Thumbnail&&)’ defaulted on its redeclaration with an exception-specification that differs from the implicit exception-specification ‘’
  116 | Thumbnail::Thumbnail(Thumbnail&& other) noexcept = default;
      | ^~~~~~~~~
../../pdf/thumbnail.cc:118:12: error: function ‘chrome_pdf::Thumbnail& chrome_pdf::Thumbnail::operator=(chrome_pdf::Thumbnail&&)’ defaulted on its redeclaration with an exception-specification that differs from the implicit exception-specification ‘’
  118 | Thumbnail& Thumbnail::operator=(Thumbnail&& other) noexcept = default;
      |            ^~~~~~~~~

Problem is that Skia SkBitmap move constructor and assignment operators are
not noexcept (and after some discussion it is going to be kept this way).

So the default implementation of move constructor and assignment operator
of pdf::Thumbnail is noexcept, as SkBitmap is a member. The fix just
removes noexcept declarations.

Bug: 819294
Change-Id: I8e6d914e4850707fbfcb51200b2a80d185cd1c19
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2411656
Reviewed-by: Jochen Eisinger <jochen@chromium.org>
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Cr-Commit-Position: refs/heads/master@{#807469}
---
 pdf/thumbnail.cc | 4 ++--
 pdf/thumbnail.h  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/pdf/thumbnail.cc b/pdf/thumbnail.cc
index d46b20ba9656..82f4790a4dc4 100644
--- a/pdf/thumbnail.cc
+++ b/pdf/thumbnail.cc
@@ -113,9 +113,9 @@ Thumbnail::Thumbnail(const gfx::Size& page_size, float device_pixel_ratio) {
   DCHECK(success);
 }
 
-Thumbnail::Thumbnail(Thumbnail&& other) noexcept = default;
+Thumbnail::Thumbnail(Thumbnail&& other) = default;
 
-Thumbnail& Thumbnail::operator=(Thumbnail&& other) noexcept = default;
+Thumbnail& Thumbnail::operator=(Thumbnail&& other) = default;
 
 Thumbnail::~Thumbnail() = default;
 
diff --git a/pdf/thumbnail.h b/pdf/thumbnail.h
index 4b2bdef99369..d62a56a50910 100644
--- a/pdf/thumbnail.h
+++ b/pdf/thumbnail.h
@@ -17,8 +17,8 @@ class Thumbnail final {
  public:
   Thumbnail();
   Thumbnail(const gfx::Size& page_size, float device_pixel_ratio);
-  Thumbnail(Thumbnail&& other) noexcept;
-  Thumbnail& operator=(Thumbnail&& other) noexcept;
+  Thumbnail(Thumbnail&& other);
+  Thumbnail& operator=(Thumbnail&& other);
   ~Thumbnail();
 
   SkBitmap& bitmap() { return bitmap_; }
-- 
2.25.1

